Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -2330,9 +2330,6 @@ - (void)messageWasSent:(TSOutgoingMessage *)message
     OWSAssert([NSThread isMainThread]);
     OWSAssert(message);
 
-    [self.editingDatabaseConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
```

Comment:
We want to save new outgoing messages "lazily" when they have been "finalized".  In the case of outgoing attachments, that means when the attachments have been (asynchronously) written to disk and their attachment ids adds to the message.  Otherwise the view may try to present them when they are incomplete.  So this should be the responsibility of `ThreadUtil` which will save messages _after_ they are finalized but _before_ they are enqueued in the outgoing message queue.